File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        memo = defaultdict(int)
        array = [[] for _ in range(len(nums) + 1)]
        for num in nums:
            memo[num] += 1
        
        for number, count in memo.items():
            array[count].append(number)
        
        res = []
        for i in range(len(array) - 1, 0, -1):
            for num in array[i]:
                res.append(num)
                if len(res) == k:
                    return res
        return []

File: Data Structures & Algorithms/jump-game/submission-0.cpp
class Solution {
public:
    bool canJump(const vector<int>& nums) {
        int n = nums.size() - 1;
        int reach = n;
        for (int i = n - 1; i >= 0; i--) {
            int max_jump = i + nums[i];
            if (max_jump >= reach) reach = i;
        }
        return reach == 0;
    }
};
